src/htmlize/patchtml.ml

```diff
@@ -20,10 +19,127 @@ let edit_html ?header ?footer ~head_childs html =
     footer
     "#footer";
   let head = html $ "head" in
-  List.iter (fun (id,child) -> 
-      opt_inserter
-        (iter (fun e -> append_child head e))
-        (Some child)
-        id)
+  List.iter (fun (id,child) ->
+    opt_inserter
+      (iter (fun e -> append_child head e))
+      (Some child)
+      id)
     head_childs;
   to_string html
+
+open Omd
+(* Keeps us from processing url links *)
+let is_url url =
+  String.equal (String.sub url 0 7) "http://" || String.equal (String.sub url 0 8) "https://"
+
+let is_table_of_content link =
+  String.equal (String.sub link 0 1) "#"
+
+(* Establish depth of given link relative to the root
+ * Then prepend the corresponding number of .. to the basename
+ * Then append index.html to fix link
+ * *)
+let handle_link link =
+  (* Escape through this branch if the link points to an http(s) address,
+   * or is a link from a Table of Contents
+   * *)
+  if is_url link.destination || is_table_of_content link.destination then
+    link
+
+  else
+    (* Establish depth *)
+    let rec get_depth acc = function
```

Comment:
You can wrap this in another function so you don't have to provide `acc` which is always 0 on first call.